lxd/instance.go

```diff
@@ -88,14 +89,36 @@ func instanceCreateFromImage(d *Daemon, r *http.Request, args db.InstanceArgs, h
 	s := d.State()
 
 	// Get the image properties.
-	_, img, err := s.DB.Cluster.GetImage(hash, db.ImageFilter{Project: &args.Project})
-	if err != nil {
-		return nil, fmt.Errorf("Fetch image %s from database: %w", hash, err)
-	}
+	var img *api.Image
+	err := s.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
+		var err error
+		_, img, err = tx.GetImageByFingerprintPrefix(ctx, hash, db.ImageFilter{Project: &args.Project})
+		if err != nil {
+			return fmt.Errorf("Fetch image %s from database: %w", hash, err)
+		}
 
-	// Set the default profiles if necessary.
-	if args.Profiles == nil {
-		args.Profiles = img.Profiles
+		// Set the default profiles if necessary.
+		if args.Profiles == nil {
+			args.Profiles = make([]api.Profile, 0, len(img.Profiles))
+			for _, name := range img.Profiles {
+				profile, err := dbCluster.GetProfileIfEnabled(ctx, tx.Tx(), args.Project, name)
```

Comment:
Same comment as above about GetProfileIfEnabled, it feels like we should avoid using this function inside a loop as it will cause duplicated project feature queries.